modules/services/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/SharingClient.kt

```diff
@@ -412,6 +435,23 @@ data class SharingRequest internal constructor(
         ) : Data {
             override fun toString() = "ClipVideo(title=${episode.title}, uuid=${episode.uuid}, start=${range.start.toSecondsWithSingleMilli()}, end=${range.end.toSecondsWithSingleMilli()})"
         }
+
+        class WebLink internal constructor(
+            override val podcast: PodcastModel? = null,
+            val textWithUrl: String,
+            val subject: String,
+        ) : Data, Sociable {
+
+            override fun sharingUrl(host: String) = textWithUrl
```

Comment:
Setting text with url here otherwise text message is not shown on certain platforms like `Twitter` and `Gmail` and I think it is important to show the message while sharing the referral link.

Reply:
It might worth in this case to split it into two function like `sharingUrl()` and `sharingText()` and combine them into one when supplying `EXTRA_TEXT`. This way you'd be still able to utilize the injected `host`.